node/resolvers/mutations/index.ts

```diff
@@ -71,88 +76,122 @@ export const Mutation = {
       vtex: { logger },
     } = ctx
 
-    const { sessionData, storefrontPermissions, segmentData } = vtex as any
-
     const settings = await checkConfig(ctx)
+    const { sessionData, storefrontPermissions, segmentData } = vtex as any
 
     checkSession(sessionData)
 
     if (!storefrontPermissions?.permissions?.includes('create-quotes')) {
       throw new GraphQLError('operation-not-permitted')
     }
 
-    const email = sessionData.namespaces.profile.email.value
-    const {
-      role: { slug },
-    } = storefrontPermissions
+    try {
+      let quoteBySeller: SellerQuoteMap = {}
 
-    const {
-      organization: { value: organizationId },
-      costcenter: { value: costCenterId },
-    } = sessionData.namespaces['storefront-permissions']
+      if (settings?.adminSetup.quotesManagedBy === 'SELLER') {
+        const sellerItems = items.filter(
+          ({ seller }) => seller && seller !== '1'
+        )
 
-    const now = new Date()
-    const nowISO = now.toISOString()
-    const expirationDate = new Date()
+        quoteBySeller = await splitItemsBySeller({
+          ctx,
+          items: sellerItems,
+        })
+      }
 
-    expirationDate.setDate(
-      expirationDate.getDate() + (settings?.adminSetup?.cartLifeSpan ?? 30)
-    )
-    const expirationDateISO = expirationDate.toISOString()
+      const hasSellerQuotes = Object.keys(quoteBySeller).length
 
-    const status = sendToSalesRep ? 'pending' : 'ready'
-    const lastUpdate = nowISO
-    const updateHistory = [
-      {
-        date: nowISO,
-        email,
+      const parentQuoteItems = hasSellerQuotes
+        ? items.filter(
+            (item) =>
```

Comment:
Isn't there a more efficient way of doing this filtering? Like, filter all items with seller === '1' (opposite of the previous filter)?

Reply:
No, because there may be a seller in the quote who:
- does not have b2b-seller-quotes installed
- does have b2b-seller-quotes installed, but has the setting to not accept quotes

In these cases, the quote remains under the management of the marketplace.